presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -454,4 +454,62 @@ public static Slice toUtf8(@SqlType(StandardTypes.VARCHAR) Slice slice)
     {
         return slice;
     }
+
+    @Description("left-padded string with the specified string for the given length")
+    @ScalarFunction
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice lpad(@SqlType(StandardTypes.VARCHAR) Slice slice, @SqlType(StandardTypes.BIGINT) long length, @SqlType(StandardTypes.VARCHAR) Slice padSlice)
```

Comment:
This implementation doesn't handle non-ASCII data correctly (the internal representation of varchars is UTF-8 byte sequences).

In particular:
* the length of the `slice` does not necessarily match the length of the unicode string if it contains any codepoints above the ASCII range.
* Padding needs to be done on a codepoint-by-codepoint basis, otherwise it can produce partial (invalid) UTF-8 sequences

For example, here's a test case that fails:

```java
assertFunction("lpad('', 1, '\u00e9')", VARCHAR, "\u00e9");
```